content/docs/rfcs/51/README.md

```diff
@@ -0,0 +1,229 @@
+shard clusters are divided into ranges:
+
+| index (range) |    usage                 |
+|    ---        |    ---                   |
+|             0 |   global                 |
+|        1 - 15 |   reserved               |
+|     16 - 1023 |   specific app protocols |
+|  1024 - 49125 |   all app protocols      |
+| 49152 - 65535 |   automatic sharding     |
+
+The informational RFC [52/WAKU2-RELAY-STATIC-SHARD-ALLOC](/spec/52) lists the current index allocations.
+
+The global shard with index 0 and the "all app protocols" range are treated in the same way,
+but choosing shards in the global cluster has a higher probability of sharing the shard with other apps.
+This offers k-anonymity and better connectivity, but comes at a higher bandwidth cost.
+
+From an app point of view, a subscription to a content topic `waku2/xxx` on a static shard would look like:
+
+`subscribe("/waku2/xxx", 43)`
+
+for global shard 43.
+And for shard 43 of the Status app (which has allocated index 16):
+
+`subscribe("/waku2/xxx", 16, 43)`
+
+## Discovery
+
+Waku v2 supports the discovery of static shards,
+so app protocols do not have to implement their own discovery method.
+To enable discovery of static shards,
+the array of shard clusters is added to [31/WAKU2-ENR](https://rfc.vac.dev/spec/31/).
+The representation is specified as follows.
+
+The array index is a 2 bytes field.
+As the array is expected to be sparse (and because ENRs do not feature an array/map type),
+the ENR contains a list of occupied array slots.
+Each shard cluster is represented by a bit vector,
+which indicates which shards of the respective shard cluster the node is part of
+(see Ethereum ENR sharding bit vector [here](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/p2p-interface.md#metadata)
+and [here](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/validator.md#sync-committee-subnet-stability)).
+
+> *Note*: We will update the [31/WAKU2-ENR](https://rfc.vac.dev/spec/31/) accordingly, once this RFC moves forward.)
+
+Having a static shard participation indication as part of the ENR allows nodes
+to discover peers that are part of shards via [33/WAKU2-DISCV5](/spec/33/) as well as via DNS.
+
+In its current raw version, this document proposes two representations in the ENR.
+(Which one to choose is open for discussion in the raw phase of the document.
+Future versions will only specify a single representation.)
+
+### One key per Shard Cluster
+
+For each shard cluster a node is part of, the node adds a separate key to its ENR.
+The representation corresponds to [Ethereum shard ENRs](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/validator.md#sync-committee-subnet-stability)).
+
+Example
+
+| key              | value                |
+|---               |---                   |
+| `relay-shard-0`  | `0x0000100000000000` |
```

Comment:
Being conscious of the size limit of the ENR, wouldn't `shard-0` be enough to describe the information?
```suggestion
| `shard-0`  | `0x0000100000000000` |
```

Reply:
I wonder about the other types of shards. Relay is the backbone of the network. Hence a relay shard, impacts all other protocols (store, light push, filter). So a relay shard sounds to be the most impactful type of shard there can be in Waku.

However, `rshard` sounds good for a more cautious and future proof name.